src/vtp/script_libs/cast_ballot_lib.py

```diff
@@ -0,0 +1,334 @@
+
+See 'cast_ballot.py -h' for usage information.
+"""
+
+# pylint: disable=wrong-import-position
+import argparse
+import logging
+import os
+import pprint
+import random
+import sys
+import traceback
+
+import pyinputplus
+
+# Local imports
+from vtp.utils.address import Address
+from vtp.utils.ballot import Ballot, BlankBallot, Contests
+from vtp.utils.common import Globals, Shellout
+
+
+class CastBallotLib:
+    """A class to wrap the cast_ballot.py script."""
+
+    def __init__(self, argv):
+        """Only to module-ize the scripts and keep things simple and idiomatic."""
+        self.argv = argv
+        self.parsed_args = None
+        self.parse_arguments()
+
+    def __str__(self):
+        """Boilerplate"""
+        return "argv=" + str(self.argv) + "\n" + "parsed_args=" + str(self.parsed_args)
+
+    def make_random_selection(self, the_ballot, the_contest):
+        """Will randomly make selections on a contest"""
+        # get the possible choices
+        choices = the_contest.get("choices")
+        tally = the_contest.get("tally")
+        # choose something randomly
+        picks = list(range(len(choices)))
+        # For plurality and max=1, the first choice is the only
+        # choice.  For plurality and max>1, the order does not matter
+        # - a selection is a selection.  For RCV, the order does
+        # matter as that is the ranking.
+        random.shuffle(picks)
+        if "plurality" == tally:
+            loop = the_contest.get("max")
+        elif "rcv" == tally:
+            loop = len(choices)
+        else:
+            raise KeyError(f"Unspoorted tally ({tally})")
+        while loop > 0:
+            the_ballot.add_selection(the_contest, picks.pop(0))
+            loop -= 1
+
+    def get_user_selection(self, the_ballot, the_contest, count, total_contests):
+        """Print the contest and get the selection(s) from the user"""
+        choices = the_contest.get("choices")
+        tally = the_contest.get("tally")
+        max_votes = the_contest.get("max")
+        # Print something
+        print(f"################ ({count} of {total_contests})")
+        print(f"Contest {the_contest.get('uid')}: {the_contest.get('name')}")
+        if tally == "plurality":
+            print(f"- This is a {tally} tally")
+            print(
+                f"- The voting is for {max_votes} open position(s)/choice(s) - "
+                "only that number of selections can be choosen."
+            )
+        else:
+            print(
+                f"- This is a {tally} tally with {max_votes} open position(s)/choice(s).  "
+                f"Regardless up to {len(choices)} selections can be rank choosen."
+            )
+
+        # Need to print the choices first up front
+        count = 0
+        for choice in choices:
+            print(f"  [{count}] {choice}")
+            count += 1
+
+        def validate_multichoice(text):
```

Comment:
Now that this not global replace move it out of a nested function into a private method.

```suggestion
    def _validate_multichoice(self, text):
```

Reply:
Actually, this is not a private method but in fact a function that is simply passed into pyinputplus.inputCustom and as such there should not be a self.  ?